### PR TITLE
Use custom resolver for libcoreclr.so also for .NET 6.0

### PR DESCRIPTION
**Pull request by henrikx**
Running into the same issue as in #889, I noticed the custom resolver code was not included for .NET 6.0 . This patch fixes the `System.DllNotFoundException` for single-file mode also in .NET 6.0

### CODE REVIEW

**Review by BDisp — APPROVED** [review]

**Comment by BDisp** [comment]
@henrikx I'm still having the `System.DllNotFoundException` error even with the `.NET 5.0`. How can I run it successful?

**Comment by henrikx** [comment]
@BDisp I am not sure about .NET 5.0, but this patch definitely fixed it for .NET 6.0

**Comment by BDisp** [comment]
But I'm still having the same error with the `.Net 6.0`. If I run my app from the `/net6.0/linux-x64/` with `dotnet ./myapp.dll` it runs without errors. But if I run it from `/net6.0/linux-x64/publish/` with `./net6.0/linux-x64/publish/myapp` it gaves me the `System.DllNotFoundException` error.

**Comment by BDisp** [comment]
@ henrikx, sorry, it working now. The reason for not working with `.Net 5.0`, was because the `Terminal.Gui` project doesn't include the `.Net 5.0`

Include this change in the `Terminal.Gui` on your PR:

`<TargetFrameworks>net472;netstandard2.0;net5.0;net6.0</TargetFrameworks>`

**Review by BDisp — COMMENTED** [review]
May be is better to change to `#if NET6_0` only and exclude the `NET5_0`.

**Comment by henrikx** [comment]
@BDisp Please check now. I have implemented the suggested changes.

**Comment by BDisp** [comment]
> @BDisp Please check now. I have implemented the suggested changes.

That is great @henrikx, thanks. Perhaps in the future, be better to keep the last two versions of the `.NET` in the projects to supports the two latest versions.

**Review by BDisp — APPROVED** [review]

**Comment by henrikx** [comment]
> Perhaps in the future, be better to keep the last two versions of the `.NET` in the projects to supports the two latest versions.

Not sure what you mean by this or in which projects you mean. This PR is good for merge now, right?

**Comment by BDisp** [comment]
> Not sure what you mean by this or in which projects you mean. This PR is good for merge now, right?

Forget what I have said before, please. The most important is to change this every time the `Terminal.Gui` project is updated with the last `.NET` version. Your PR is good to merge but you have to wait for the approval and merging by @tig, because I'm only a contributor.

**Pull request merged**
